rep-2001.rst

```diff
@@ -33,15 +33,63 @@ Specification
 End-user entry points
 ---------------------
 
-We define three main entry points for ROS users.
+We define these main entry points for ROS users.
 
  * desktop (recommended)
  * ros_base
  * ros_core
+ * ros_cpp_core (advanced)
```

Comment:
Somewhat pedantic, but with `ros_core`, I think `ros_core_cpp` would follow the naming convention we generally use (see all packages below).

Also the "advanced" tag is a little unclear (what's advanced about it? Isn't it more basic than `ros_core`? etc). I might change it to `barebones` if you'd like to include some qualifier, which I'm not sure is necessary. The name itself is very telling.